packages/protocol/test/common/goldtoken.ts

```diff
@@ -51,6 +56,59 @@ contract('GoldToken', (accounts: string[]) => {
     })
   })
 
+  describe('#burn()', () => {
+    let startBurn: BigNumber
+
+    beforeEach(async () => {
+      startBurn = await goldToken.getBurnedAmount()
+    })
+
+    it('burn address starts with zero balance', async () => {
+      assertEqualBN(await goldToken.balanceOf(burnAddress), 0)
+    })
+
+    it('burn starts as start burn amount', async () => {
+      assertEqualBN(await goldToken.getBurnedAmount(), startBurn)
+    })
+
+    it('Burned amount equals the balance of the burn address', async () => {
+      assertEqualBN(await goldToken.getBurnedAmount(), await goldToken.balanceOf(burnAddress))
+    })
+
+    it('returns right burned amount', async () => {
+      await goldToken.burn(ONE_GOLDTOKEN)
+
+      assertEqualBN(await goldToken.getBurnedAmount(), ONE_GOLDTOKEN.plus(startBurn))
+    })
+  })
+
+  describe('#circulatingSupply()', () => {
+    let mockGoldToken: MockGoldTokenInstance
+
+    beforeEach(async () => {
+      mockGoldToken = await MockGoldToken.new()
+      // set supply to 1K
+      await mockGoldToken.setTotalSupply(ONE_GOLDTOKEN.multipliedBy(1000))
+    })
+
+    it('matches circulatingSupply() when there was no burn', async () => {
+      assertEqualBN(await mockGoldToken.circulatingSupply(), await mockGoldToken.totalSupply())
+    })
+
+    it('decreases when there was a burn', async () => {
+      // mock a burn
+      await mockGoldToken.setBalanceOf(burnAddress, ONE_GOLDTOKEN)
```

Comment:
Is there any reason why goldToken.burn(ONE_GOLDTOKEN) is not called here instead of setTotalSupply ?

Reply:
The reason is that I'm using mockGold here rather than GoldToken, because as Celo is a native token couldn't figure out a way to manipulate the supply from truffle.